lib/diagnostic.js

```diff
@@ -5,7 +5,20 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
+const min = function (previous, current) {
+  const workingArray = previous.slice();
```

Comment:
```previous``` and ```current``` are assumed to be numbers.  Since you're trying to invoke ```slice()```, an array method, on a number, this will throw an error.